server/src/internalClusterTest/java/org/opensearch/gateway/remote/RemoteRoutingTableServiceIT.java

```diff
@@ -0,0 +1,324 @@
+            pathHashAlgo
+        );
+    }
+
+    private void verifyUpdatesInManifestFile(RemoteManifestManager remoteManifestManager) {
+        Optional<ClusterMetadataManifest> latestManifest = remoteManifestManager.getLatestClusterMetadataManifest(
+            getClusterState().getClusterName().value(),
+            getClusterState().getMetadata().clusterUUID()
+        );
+        assertTrue(latestManifest.isPresent());
+        ClusterMetadataManifest manifest = latestManifest.get();
+        assertTrue(manifest.getDiffManifest().getIndicesRoutingUpdated().contains(INDEX_NAME));
+        assertTrue(manifest.getDiffManifest().getIndicesDeleted().isEmpty());
+        assertFalse(manifest.getIndicesRouting().isEmpty());
+        assertEquals(1, manifest.getIndicesRouting().size());
+        assertTrue(manifest.getIndicesRouting().get(0).getUploadedFilename().contains(indexRoutingPath.buildAsString()));
+    }
+
+    private List<RoutingTable> getRoutingTableFromAllNodes() throws ExecutionException, InterruptedException {
+        String[] allNodes = internalCluster().getNodeNames();
+        List<RoutingTable> routingTables = new ArrayList<>();
+        for (String node : allNodes) {
+            RoutingTable routingTable = internalCluster().client(node)
+                .admin()
+                .cluster()
+                .state(new ClusterStateRequest().local(true))
+                .get()
+                .getState()
+                .routingTable();
+            routingTables.add(routingTable);
+        }
+        return routingTables;
+    }
+
+    private boolean areRoutingTablesSame(List<RoutingTable> routingTables) {
+        if (routingTables == null || routingTables.isEmpty()) {
+            return false;
+        }
+
+        RoutingTable firstRoutingTable = routingTables.get(0);
+        for (RoutingTable routingTable : routingTables) {
+            if (!compareRoutingTables(firstRoutingTable, routingTable)) {
+                logger.info("Responses are not the same: {} {}", firstRoutingTable, routingTable);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean compareRoutingTables(RoutingTable a, RoutingTable b) {
+        if (a == b) return true;
+        if (b == null || a.getClass() != b.getClass()) return false;
+        if (a.version() != b.version()) return false;
+        if (a.indicesRouting().size() != b.indicesRouting().size()) return false;
+
+        for (Map.Entry<String, IndexRoutingTable> entry : a.indicesRouting().entrySet()) {
+            IndexRoutingTable thisIndexRoutingTable = entry.getValue();
+            IndexRoutingTable thatIndexRoutingTable = b.indicesRouting().get(entry.getKey());
+            if (thatIndexRoutingTable == null || !compareIndexRoutingTables(thisIndexRoutingTable, thatIndexRoutingTable)) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
Do we need this custom logic? Why cant we use IndexRoutingTable.equals?

Reply:
Used  IndexRoutingTable.equals.